### ISSUE TITLE
Container node using a lot of CPU

### ISSUE THREAD

**Issue by lsena** [vespa-engine/vespa#22875]
**Describe the bug**
I've been doing some benchmarks to understand the best strategy for production deployment of Vespa.
I'm stress testing the ANN capabilities of Vespa with the following setup:

- Container machine: 8 CPUs / 16GB RAM
- Content machine: 2 CPUs / 16GB RAM
- 1.5M dense vectors with 768 dim (float)

The result is that during the benchmark, the container machine reaches ~100% CPU while the content machine stays below 80%.

Some screenshots:
<img width="1568" alt="image" src="https://user-images.githubusercontent.com/19229049/171645338-759ac1b5-bdd2-494d-868b-43b16ffbc327.png">

services.xml

```
<services version="1.0">
    <admin version="2.0">
        <configservers>
            <configserver hostalias="node0"/>
        </configservers>
        <cluster-controllers>
            <cluster-controller hostalias="node0"/>
        </cluster-controllers>
        <slobroks>
            <slobrok hostalias="node0"/>
        </slobroks>
        <adminserver hostalias="node0"/>
    </admin>

    <container id="stateless-container-cluster" version="1.0">
        <search/>
        <document-api/>
        <nodes>
            <jvm options="-Xms8g -Xmx8g"/>
            <node hostalias="node1"/>
        </nodes>
    </container>
    <content id="ecom" version="1.0">
        <documents>
            <document type="ecom" mode="index"></document>
        </documents>
        <redundancy>2</redundancy>
        <engine>
            <proton>
                <tuning>
                    <searchnode>
                        <summary>
                            <io>
                                <read>directio</read>
                            </io>
                            <store>
                                <cache>
                                    <maxsize-percent>10</maxsize-percent>
                                </cache>
                            </store>
                        </summary>
                    </searchnode>
                </tuning>
                <searchable-copies>2</searchable-copies>
            </proton>
        </engine>
        <group>
            <distribution partitions="1|*"/>
            <group name="group0" distribution-key="0">
                <node hostalias="node0" distribution-key="0"/>
            </group>
            <group name="group2" distribution-key="2">
                <node hostalias="node2" distribution-key="2"/>
            </group>
        </group>
    </content>
</services>
```

schema

```
schema ecom {
    document ecom {
        field id type string {
            indexing: summary | attribute
            attribute {
                fast-search
            }
        }
        field title type string {
            indexing: index
            index: enable-bm25
        }
        field description type string {
            indexing: summary
        }
        field imUrl type string {
            indexing: attribute
        }
        field brands type array<string> {
            indexing: index | attribute
        }
        field stock_size_10 type int {
            indexing: attribute
            attribute {
                fast-search
            }
        }
        field embedding type tensor<float>(d0[768]) {
            indexing: attribute | index
            attribute {
                distance-metric: angular
            }
            index {
                hnsw {
                    max-links-per-node: 12
                    neighbors-to-explore-at-insert: 500
                }
            }
        }
    }
    document-summary id {
        summary id type string { source: id}
    }
    rank-profile semantic inherits default {
        first-phase {
            expression: closeness(field, embedding)
        }
    }
}
```

query:
```
yql = f"select id from sources ecom where ([{{'approximate':true, 'targetHits': 10 }}]nearestNeighbor(embedding, query_embedding));"
query = {
    "yql": yql,
    "hits": 10,
    "presentation.timing": True,
    'presentation.summary': 'id',
    "ranking.features.query(query_embedding)": computed_embedding,
    "ranking.profile": "semantic"
}
```

container node vespa.log:
```
1654163378.893616       xx.xx.xx.xx     690     container       stdout  info    [64021.851s][info][gc] GC(2087) Pause Young (Normal) (GCLocker Initiated GC) 5218M->302M(8192M) 5.326ms
1654163380.677056       xx.xx.xx.xx     690     container       stdout  info    [64023.646s][info][gc] GC(2088) Pause Young (Normal) (G1 Evacuation Pause) 5202M->302M(8192M) 5.154ms
1654163382.400767       xx.xx.xx.xx     690     container       stdout  info    [64025.358s][info][gc] GC(2089) Pause Young (Normal) (G1 Evacuation Pause) 5202M->303M(8192M) 5.217ms
1654163384.092301       xx.xx.xx.xx     690     container       stdout  info    [64027.061s][info][gc] GC(2090) Pause Young (Normal) (G1 Evacuation Pause) 5203M->303M(8192M) 5.430ms
1654163385.816428       xx.xx.xx.xx     690     container       stdout  info    [64028.773s][info][gc] GC(2091) Pause Young (Normal) (GCLocker Initiated GC) 5207M->303M(8192M) 5.002ms
1654163387.503429       xx.xx.xx.xx     690     container       stdout  info    [64030.473s][info][gc] GC(2092) Pause Young (Normal) (G1 Evacuation Pause) 5203M->303M(8192M) 4.980ms
1654163389.254053       xx.xx.xx.xx     690     container       stdout  info    [64032.223s][info][gc] GC(2093) Pause Young (Normal) (G1 Evacuation Pause) 5203M->303M(8192M) 4.989ms
1654163391.072562       xx.xx.xx.xx     690     container       stdout  info    [64034.030s][info][gc] GC(2094) Pause Young (Normal) (GCLocker Initiated GC) 5211M->303M(8192M) 5.124ms
1654163392.793819       xx.xx.xx.xx     690     container       stdout  info    [64035.763s][info][gc] GC(2095) Pause Young (Normal) (G1 Evacuation Pause) 5203M->303M(8192M) 4.707ms
1654163394.511580       xx.xx.xx.xx     690     container       stdout  info    [64037.481s][info][gc] GC(2096) Pause Young (Normal) (G1 Evacuation Pause) 5203M->303M(8192M) 4.803ms
1654163396.277132       xx.xx.xx.xx     690     container       stdout  info    [64039.246s][info][gc] GC(2097) Pause Young (Normal) (G1 Evacuation Pause) 5203M->303M(8192M) 5.429ms
1654163398.018148       xx.xx.xx.xx     690     container       stdout  info    [64040.987s][info][gc] GC(2098) Pause Young (Normal) (G1 Evacuation Pause) 5203M->303M(8192M) 5.225ms
1654163399.749843       xx.xx.xx.xx     690     container       stdout  info    [64042.719s][info][gc] GC(2099) Pause Young (Normal) (G1 Evacuation Pause) 5203M->303M(8192M) 4.714ms
1654163401.486457       xx.xx.xx.xx     690     container       stdout  info    [64044.456s][info][gc] GC(2100) Pause Young (Normal) (G1 Evacuation Pause) 5203M->303M(8192M) 4.806ms
1654163403.205619       xx.xx.xx.xx     690     container       stdout  info    [64046.163s][info][gc] GC(2101) Pause Young (Normal) (G1 Evacuation Pause) 5207M->303M(8192M) 3.852ms
1654163404.929094       xx.xx.xx.xx     690     container       stdout  info    [64047.898s][info][gc] GC(2102) Pause Young (Normal) (G1 Evacuation Pause) 5207M->303M(8192M) 3.908ms
1654163406.657463       xx.xx.xx.xx     690     container       stdout  info    [64049.627s][info][gc] GC(2103) Pause Young (Normal) (G1 Evacuation Pause) 5211M->303M(8192M) 2.740ms
```
I get one of those lines every 1-2s.
Even though it's doing 2.5k Q/s, 5GB allocations every 1-2s seems excessive.

**To Reproduce**
1. Setup 2 separate machines (1 content and 1 container)
2. Load application.zip (need to edit hosts file first)
3. Ingest vectors (unfortunately, it's a bit difficult to provide my dataset atm, but I'm certain this will happen with any other public dataset)
4. Run N parallel clients with the provided query (I'm using pyvespa)

**Expected behavior**
I expected the content nodes to consume more CPU than the container nodes since they do the bulk of the work (cosine distance, graph transversal, etc).

**Environment (please complete the following information):**
 - OS (Docker Host):
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=21.10
DISTRIB_CODENAME=impish
DISTRIB_DESCRIPTION="Ubuntu 21.10"
 - Infrastructure: Azure VMs. Content: E2ads v5. Container: F8s v2

**Vespa version**
docker hub latest (7.589.25)

**Additional context**
[application.zip](https://github.com/vespa-engine/vespa/files/8824694/application.zip)

**Comment by jobergum** [comment]
You could try to see if you get any improvement from changing from G1 to ParallelGC - but it's expected that at 2500 requests/s with large query tensors you will see heavy GC, as also can be seen from the traces the G1GC works well. It might also be that `nproc` returns a higher number than 8 on  Azure VMs?

From [Vespa container tuning](https://docs.vespa.ai/en/performance/container-tuning.html)
> The thread pools are by default scaled on the system resources as reported by the JVM (Runtime.getRuntime().availableProcessors()). It's paramount that the -XX:ActiveProcessorCount/jvm_availableProcessors configuration is correct for the container to work optimally. 

```
gc-options="-Xms8g -Xmx8g" options=" -XX:ActiveProcessorCount=8"
```

Please use `gc-options` and not `options` - former is for gc related settings.

**Comment by jobergum** [comment]
https://github.com/vespa-engine/vespa/pull/23248 and https://github.com/vespa-engine/vespa/pull/23253 improve CPU usage related to getting JSON queries across by approximately 2x, meaning you can get 2x more queries through per v-cpu. Available in 8.9.18 or larger, not published yet, but it's coming to https://hub.docker.com/r/vespaengine/vespa/tags soon.
